src/main/kotlin/no/nav/syfo/api/endpoints/VedtakEndpoints.kt

```diff
@@ -67,5 +68,21 @@ fun Route.registerVedtakEndpoints(
 
             call.respond(HttpStatusCode.Created, VedtakResponseDTO.createFromVedtak(vedtak = newVedtak))
         }
+        post(ferdigPath) {
```

Comment:
`POST` pleier å gjenspeile at en helt ny ressurs blir laget. Hadde det vært mer semantisk riktig å bruke `PUT` eller `PATCH` her? 😊 

Synes det kan være greit å prøve å følge REST konvensjoner så godt vi kan slik at API'ene våre blir oversiktlige -> https://restfulapi.net/resource-naming/

Sånn ideelt sett skulle kanskje API'et vårt sett slik ut, men vi får prøve oss på dette andre steder.
`GET /vedtak/{UUID}` - Hent aktuelt vedtak
`PUT /vedtak/{UUID}` - Endre aktuelt vedtak
`POST /vedtak` - Lag nytt vedtak

Reply:
Føler det hadde vært naturlig å angi uuid til vedtaket man ønsker å ferdigstille som del av APIet her. Så `POST /vedtak/{uuid}/ferdigstill` (evt `PUT`)

Reply:
Slik jeg har forstått det skal REST verbet beskrive selve handlingen og man skal egentlig ikke bruke verb som en del av URL'en (For eks. `/ferdigstill`). Så tror det som hadde vært mest riktig her er `PUT/PATCH /vedtak/{UUID}`

(Se seksjon 3 i [linken over](https://restfulapi.net/resource-naming/))

Reply:
Men hvis man har PUT/PATCH /vedtak/{UUID} uten å angi noe verb så må tjenesten nesten ta inn en dto. Og da blir det uklart hva intensjonen er: Enten tillate alle mulige endringer (noe som gjør tjenesten mer komplisert og åpner for endringer som vi antakelig ikke ønsker), evnt likevel begrense den til å bare kunne ferdigbehandle - men da synes jeg tjenestens innhold blir misvisende i forhold til path'en.

Reply:
2 alternativer kan være å enten motta selve objekt dto'en hvor vi sjekker hvilken endring det gjelder og returnerer 400 hvis det ikke er en gyldig endring. Eller en enum som beskriver endringen, med en payload som er endringen.
Alternativ 2 kunne sett noe slikt ut: (Med eksempel for endring av veileder)
```kotlin
sealed class Update {
	class FerdigstillVedtak() : Update()
	class ChangeVeileder(veilederIdent: String): Updated()
}
```